pkg/azureclients/loadbalancerclient/azure_loadbalancerclient.go

```diff
@@ -256,29 +256,42 @@ func (c *Client) createOrUpdateLB(ctx context.Context, resourceGroupName string,
 		"Microsoft.Network/loadBalancers",
 		loadBalancerName,
 	)
-	decorators := []autorest.PrepareDecorator{
-		autorest.WithPathParameters("{resourceID}", map[string]interface{}{"resourceID": resourceID}),
-		autorest.WithJSON(parameters),
-	}
-	if etag != "" {
-		decorators = append(decorators, autorest.WithHeader("If-Match", autorest.String(etag)))
+	resp, rerr := c.armClient.GetResource(ctx, resourceID, "")
```

Comment:
could we avoid querying requests in the write API? Generally, the GET requests should be controlled by the reconciling loop and hence caches could be used to reduce number of API calls.